Xamarin.Essentials/Calendars/Calendars.android.cs

```diff
@@ -618,26 +631,27 @@ static RecurrenceRule GetRecurranceRuleForEvent(string rule)
                 }
             }
 
-            if (rule.Contains("BYMONTHDAY="))
+            if (rule.Contains(byMonthDaySearch))
             {
-                var ruleOccurenceMonthDays = rule.Substring(rule.IndexOf("BYMONTHDAY=", StringComparison.Ordinal) + 11);
+                var ruleOccurenceMonthDays = rule?.Substring(rule.IndexOf(byMonthDaySearch, StringComparison.Ordinal) + byMonthDaySearch.Length);
                 ruleOccurenceMonthDays = ruleOccurenceMonthDays.Contains(";") ? ruleOccurenceMonthDays.Substring(0, ruleOccurenceMonthDays.IndexOf(";", StringComparison.Ordinal)) : ruleOccurenceMonthDays;
-                recurranceRule.DayOfTheMonth = (uint)Math.Abs(Convert.ToInt32(ruleOccurenceMonthDays.Split(',').FirstOrDefault()));
+                uint.TryParse(ruleOccurenceMonthDays.Split(',').FirstOrDefault(), out var result);
+                recurranceRule.DayOfTheMonth = result;
             }
 
-            if (rule.Contains("BYMONTH="))
+            if (rule.Contains(byMonthSearch))
```

Comment:
This needs to be s StringComparison.Ordinal contains, similar to the IndexOf below

Reply:
Altered accordingly